### PR TITLE
Docs: Add documentation for library elements API

### LINKED ISSUE #35575: Docs: Add documentation for library-elements API
**What would you like to be added**:
Currently there's no documentation for the library-elements API, which developers might like to take advantage of.

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #35575

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/http_api/library_element.md [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+- **401** – Unauthorized
+- **403** – Access Denied
+- **404** – Library element not found
+- **412** – Version mismatch
+
+## Delete library element
+
+`DELETE /api/library-elements/:uid`
+
+Deletes an existing library element identified by UID. Will not succeed if the library element is connected. This operation cannot be reverted.
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:424]:
```suggestion
Deletes an existing library element as specified by the UID. This operation cannot be reverted.

>**Note:** You cannot delete a library element that is connected. This operation cannot be reverted.
```

File: docs/sources/http_api/library_element.md [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+            }
+        }
+    }
+}
+```
+
+Status Codes:
+
+- **200** – Updated
+- **400** – Errors (name or uid already exists, invalid json, missing or invalid fields, etc.)
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:414]:
```suggestion
- **400** – Errors (for example, name or UID already exists, invalid JSON, missing or invalid fields, and so on).
```

File: docs/sources/http_api/library_element.md [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+- **401** – Unauthorized
+- **403** – Access Denied
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:416]:
```suggestion
- **403** – Access denied
```

File: docs/sources/http_api/library_element.md [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+
+## Update library element
+
+`PATCH /api/library-elements/:uid`
+
+Updates an existing library element identified by uid.
+
+JSON Body schema:
+
+- **folderId** – The folder id of the folder where the library element is stored.
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:347]:
```suggestion
- **folderId** – ID of the folder where the library element is stored.
```

File: docs/sources/http_api/library_element.md [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+- **name** – The name of the library element.
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:348]:
```suggestion
- **name** –Name of the library element.
```

File: docs/sources/http_api/library_element.md [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+- **model** – The json model for the library element.
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:349]:
```suggestion
- **model** – The JSON model for the library element.
```

File: docs/sources/http_api/library_element.md [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+- **kind** – Kind of element to create, `1` for library panels or `2` for library variables.
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:350]:
```suggestion
- **kind** – Kind of element to create. Use `1` for library panels or `2` for library variables.
```

File: docs/sources/http_api/library_element.md [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+- **version** – Version of the library element being updated.
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:351]:
```suggestion
- **version** – Version of the library element you are updating.
```

File: docs/sources/http_api/library_element.md [hunk 9]
```diff
@@ -0,0 +1,453 @@
...
+- **uid** – Optional [unique identifier](/http_api/library_element/#identifier-id-vs-unique-identifier-uid).
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:352]:
```suggestion
- **uid** – Optional, the [unique identifier](/http_api/library_element/#identifier-id-vs-unique-identifier-uid).
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

File: docs/sources/http_api/library_element.md [hunk 10]
```diff
@@ -0,0 +1,453 @@
...
+
+Creates a new library element.
+
+JSON Body schema:
+
+- **folderId** – The folder id of the folder where the library element is stored.
+- **name** – The name of the library element.
+- **model** – The json model for the library element.
+- **kind** – Kind of element to create, `1` for library panels or `2` for library variables.
+- **uid** – Optional [unique identifier](/http_api/library_element/#identifier-id-vs-unique-identifier-uid).
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:274]: Make similar changes suggested in other review comments.

File: docs/sources/http_api/library_element.md [hunk 11]
```diff
@@ -0,0 +1,453 @@
...
+            }
+        }
+    }
+}
+```
+
+Status Codes:
+
+- **200** – Created
+- **400** – Errors (name or uid already exists, invalid json, missing or invalid fields, etc.)
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:335]:
```suggestion
- **400** – Errors (for example, name or UID already exists, invalid JSON, missing or invalid fields, and so on).
```

File: docs/sources/http_api/library_element.md [hunk 12]
```diff
@@ -0,0 +1,453 @@
...
+- **401** – Unauthorized
+- **403** – Access Denied
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:337]:
```suggestion
- **403** – Access denied
```

**Review by achatterjee-grafana — COMMENTED** [review]
Some more change request.

File: docs/sources/http_api/library_element.md [hunk 13]
```diff
@@ -0,0 +1,453 @@
...
+Query parameters:
+
+- **searchString** – Part of the name or description searched for
+- **kind** – Kind of element to search for, `1` for library panels or `2` for library variables
+- **sortDirection** – Sort order of elements, `alpha-asc` for ascending or `alpha-desc` for descending
+- **typeFilter** – Comma separated list of types to filter elements by
+- **excludeUid** – Element UID to exclude from results
+- **folderFilter** – Comma separated list of folder id's to filter elements by
+- **perPage** – Limit the number of returned results per page (default 100)
+- **page** – Use this parameter to access hits beyond `perPage`. Numbering starts at 1. `perPage` param acts as page size.
```
[the remark this answers is not in the document]
Comment (hugohaggmark) [reply docs/sources/http_api/library_element.md:33]: I think this part is important `Numbering starts at 1. 'perPage' param acts as page size.`

File: docs/sources/http_api/library_element.md [hunk 14]
```diff
@@ -0,0 +1,453 @@
...
+Status Codes:
+
+- **200** – Found
+- **401** – Unauthorized
+
+## Get library element by uid
+
+`GET /api/library-elements/:uid`
+
+Will return the library-element given the uid.
```
[the remark this answers is not in the document]
Comment (hugohaggmark) [reply docs/sources/http_api/library_element.md:101]: This endpoint returns a library element with the supplied UID.

File: docs/sources/http_api/library_element.md [hunk 15]
```diff
@@ -0,0 +1,453 @@
...
+
+- **200** – Found
+- **401** – Unauthorized
+- **404** – Library element not found
+
+## Get library element by name
+
+`GET /api/library-elements/name/:name`
+
+Will return the library-element given the name.
```
[the remark this answers is not in the document]
Comment (hugohaggmark) [reply docs/sources/http_api/library_element.md:161]: This endpoint returns a library element with the given name.

File: docs/sources/http_api/library_element.md [hunk 16]
```diff
@@ -0,0 +1,453 @@
...
+
+- **200** – Found
+- **401** – Unauthorized
+- **404** – Library element not found
+
+## Get library element connections
+
+`GET /api/library-elements/:uid/connections`
+
+Will return a list of connections for a library-element given the uid.
```
[the remark this answers is not in the document]
Comment (hugohaggmark) [reply docs/sources/http_api/library_element.md:221]: I think UID is an important piece of information in this context.

**Comment by hugohaggmark** [comment]
Thanks for all the great feedback @achatterjee-grafana! I've committed most of it but there are some places that you need to revisit. Thank you!

File: docs/sources/http_api/library_element.md [hunk 17]
```diff
@@ -0,0 +1,455 @@
...
+
+Query parameters:
+
+- **searchString** – Part of the name or description searched for
+- **kind** – Kind of element to search for. Use `1` for library panels or `2` for library variables.
+- **sortDirection** – Sort order of elements. Use `alpha-asc` for ascending and `alpha-desc` for descending sort order.
+- **typeFilter** – A comma separated list of types to filter the elements by.
+- **excludeUid** – Element UID to exclude from search results.
+- **folderFilter** – A comma separated list of folder ID(s) to filter the elements by.
+- **perPage** – Limit the number of results per page; default is 100.
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/library_element.md:32]:
```suggestion
- **perPage** – The number of results per page; default is 100.
```

File: docs/sources/http_api/library_element.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (achatterjee-grafana) [reply docs/sources/http_api/library_element.md:33]: Suggestion:
The page for a set of records, given that only `perPage` records are returned at a time. Numbering starts at 1.

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review based on @hugohaggmark's feedback.

**Review by achatterjee-grafana — APPROVED** [review]
Approving, based on the fact that we should be good to go once you accept the review comments. TY!

**Pull request merged**